### PR TITLE
Use latest build_config in build_runner

### PR DESCRIPTION
**Pull request by natebosch**
- Add a dependency override to use it before it's published.
- Fix some errors in the config in test code. Drop unnecessary targets
  config.

### CODE REVIEW

File: build_config/lib/src/build_config.dart [hunk 1]
```diff
@@ -145,24 +145,27 @@ Map<String, T> _normalizeBuilderDefinitions<T>(
...
-  var yamlKey = yamlMap.nodes.keys.singleWhere(
-      (k) => (k as YamlScalar).value == err.key,
-      orElse: () => null) as YamlScalar;
-
   var message = 'Could not create `${err.className}`.';
-  if (yamlKey == null) {
-    assert(err.key == null);
-    message = '${yamlMap.span.message(message)} ${err.innerError}';
+  if (err.key == null) {
+    if (err.map is YamlMap) {
```
Comment (kevmoo) [inline build_config/lib/src/build_config.dart]: nit: if you pull out the `map` as a local var and do the type check there, you shouldn't have to cast it below.

**Comment by natebosch** [comment]
> Don't see new test coverage for the new/better pretty things

It's an exception path which is little used. We don't generally test those.

> I don't understand how the changes to the test files map to the changes in the code.

These are changes required because the latest `build_config` is more strict than it used to be. The invalid config fails at parse time now rather than later at runtime.

File: build_config/lib/src/build_config.dart (hunk 1 above)
Comment (natebosch) [reply build_config/lib/src/build_config.dart]: Done

File: build_config/lib/src/build_config.dart [hunk 2]
```diff
@@ -145,24 +145,27 @@ Map<String, T> _normalizeBuilderDefinitions<T>(
...
       message += '${err.message}\n';
     } else {
       message += '${err.innerError}\n';
     }
-    message = yamlKey.span.message(message);
+    if (map is YamlMap) {
+      var yamlKey = map.nodes.keys.singleWhere(
+          (k) => (k as YamlScalar).value == err.key,
+          orElse: () => null) as YamlScalar;
+      message += yamlKey.span.message(message);
```
Comment (jakemac53) [inline build_config/lib/src/build_config.dart]: are you sure you want the += here? I think the yaml `.span.message` call already incorporates the message you pass in.

File: build_config/lib/src/build_config.dart [hunk 3]
```diff
@@ -145,24 +145,27 @@ Map<String, T> _normalizeBuilderDefinitions<T>(
...
     if (err.message != null) {
       message += '${err.message}\n';
     } else {
       message += '${err.innerError}\n';
     }
-    message = yamlKey.span.message(message);
+    if (map is YamlMap) {
+      var yamlKey = map.nodes.keys.singleWhere(
+          (k) => (k as YamlScalar).value == err.key,
+          orElse: () => null) as YamlScalar;
```
Comment (jakemac53) [inline build_config/lib/src/build_config.dart]: nit: if adding an `orElse` which returns `null`, then we should handle the `null` case.

**Comment by natebosch** [comment]
Holding off on the changes to the exception handling. The important thing here is that the test works with the stricter parsing and I had only changed the exception handling to get actionable errors from the test code.

**Comment by natebosch** [comment]
Hmm, analyzer warnings point to a potential usage that we've broken with the latest build_config.

@jakemac53 can you take a look? Technically this isn't a usage pattern we expected people (outside of our tests) to use, but we did change the public surface area when we changed the signature of some of the constructors.

https://travis-ci.org/dart-lang/build/jobs/383788477

**Comment by jakemac53** [comment]
https://github.com/dart-lang/build/pull/1500 now does this plus fixing the other tests and some other minor tweaks that were needed for that

**Pull request closed without merging**